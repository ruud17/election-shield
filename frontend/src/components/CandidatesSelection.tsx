import { FC, ChangeEvent } from "react";
import { Row, Alert, Form, Card, Col } from "react-bootstrap";
import { Candidates } from "../utils/enum";
import DonaldImg from "../images/Donald.jpeg";
import JoeImg from "../images/Joe.jpeg";

export interface CandidatesSelectionProps {
  selectedCandidate: string;
  handleOptionChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

const CandidatesSelection: FC<CandidatesSelectionProps> = ({
  selectedCandidate,
  handleOptionChange,
}) => {
  return (
    <Row className="candidate">
      <Alert variant="info">
        Please select your preferred presidential candidate
      </Alert>
      <Col>
        <Form.Group>
          <Card className="bg-light text-black">
            <Card.Img variant="top" src={DonaldImg} />
            <Card.Body>
              <Card.Title>
                <Form.Check
                  type="radio"
                  label={Candidates.DonaldTrump}
                  value={Candidates.DonaldTrump}
                  checked={selectedCandidate === Candidates.DonaldTrump}
                  onChange={handleOptionChange}
                />
              </Card.Title>
            </Card.Body>
          </Card>
        </Form.Group>
      </Col>
      <Col>
        <Form.Group>
          <Card className="bg-light text-black">
            <Card.Img variant="top" src={JoeImg} />
            <Card.Body>
              <Card.Title>
                <Form.Check
                  type="radio"
                  label={Candidates.JoeBiden}
                  value={Candidates.JoeBiden}
                  checked={selectedCandidate === Candidates.JoeBiden}
                  onChange={handleOptionChange}
                />
              </Card.Title>
            </Card.Body>
          </Card>
        </Form.Group>
      </Col>
    </Row>
  );
};

export default CandidatesSelection;
