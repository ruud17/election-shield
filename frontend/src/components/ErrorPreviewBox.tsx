import { FC } from "react";
import { Row, Alert } from "react-bootstrap";
import { ViewError } from "../utils/error";

interface ErrorPreviewBoxProps {
  errors: unknown[];
}

const ErrorPreviewBox: FC<ErrorPreviewBoxProps> = ({ errors }) => {
  return (
    <Row>
      {errors.map((err, ind) => (
        <Alert variant="danger" key={ind}>
          <p className="mb-0">{ViewError.makeNewErrorToDisplay(err)}</p>
        </Alert>
      ))}
    </Row>
  );
};

export default ErrorPreviewBox;
