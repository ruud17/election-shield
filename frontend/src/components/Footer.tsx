import { FC } from "react";
import { Container, Navbar } from "react-bootstrap";

const Footer: FC = () => {
  return (
    <Navbar fixed="bottom" bg="dark" variant="dark" className="footer">
      <Navbar.Text>&copy; {new Date().getFullYear()} Mehmed Jusic</Navbar.Text>
    </Navbar>
  );
};

export default Footer;
