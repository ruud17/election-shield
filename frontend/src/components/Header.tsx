import { FC } from "react";
import { Navbar, Image } from "react-bootstrap";
import VoteLogo from "../images/vote.svg";

const Header: FC = () => {
  return (
    <Navbar fixed="top" bg="dark" data-bs-theme="light" className="header">
      <Image src={VoteLogo} className="election-logo " />
      Election Shield
    </Navbar>
  );
};

export default Header;
