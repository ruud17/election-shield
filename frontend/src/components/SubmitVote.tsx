import { FC } from "react";
import { Container, Row, Col, Button, Form } from "react-bootstrap";

const SubmitVote: FC = () => {
  return (
    <Row className="submit-vote">
      <Col>
        <Form.Check type="checkbox">
          <Form.Check.Input type="checkbox" />
          <Form.Check.Label>
            Accept Terms of Service and Data Collection Consent
          </Form.Check.Label>
        </Form.Check>
      </Col>
      <Col className="d-flex justify-content-end">
        <Button variant="primary" size="sm">
          Submit
        </Button>
      </Col>
    </Row>
  );
};

export default SubmitVote;
