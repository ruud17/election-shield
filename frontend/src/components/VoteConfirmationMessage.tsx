import { FC } from "react";
import { Container } from "react-bootstrap";

interface VoteConfirmationMessageProps {
  // name: string;
}

const VoteConfirmationMessage: FC<VoteConfirmationMessageProps> = () => {
  return <Container>Thank you for successfull voting!!!</Container>;
};

export default VoteConfirmationMessage;
