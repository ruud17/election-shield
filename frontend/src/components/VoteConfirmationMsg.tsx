import { FC } from "react";
import { Row, Alert, Image } from "react-bootstrap";
import SuccessVoteIcon from "../images/success_vote.svg";

const VoteConfirmationMsg: FC = () => {
  return (
    <div className="success-vote">
      <Row>
        <Image src={SuccessVoteIcon} />
      </Row>
      <Row>
        <Alert variant="success">
          <Alert.Heading>You have successfully voted</Alert.Heading>
          <hr />
          <p className="mb-0">Thank you!</p>
        </Alert>
      </Row>
    </div>
  );
};

export default VoteConfirmationMsg;
