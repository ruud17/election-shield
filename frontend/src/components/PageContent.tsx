import { FC, useState, ChangeEvent } from "react";
import { Container } from "react-bootstrap";
import VoterDetailsForm from "./VoterDetailsForm";
import CandidatesSelection from "./CandidatesSelection";
import SubmitVote from "./SubmitVote";
import { VoterDetails } from "../utils/interfaces";
import { initialVoterDetailsData } from "../utils/defaultStates";
import { Candidates } from "../utils/enum";

const PageContent: FC = () => {
  const [voterDetails, setVoterDetails] = useState<VoterDetails>(
    initialVoterDetailsData
  );

  const [selectedCandidate, setSelectedCandidate] = useState<string>(
    Candidates.DonaldTrump
  );

  const [isTermsCheckboxConfirmed, setIsTermsCheckboxConfirmed] =
    useState<boolean>(false);

  const handleVoterDetailChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target;
    setVoterDetails({
      ...voterDetails,
      [name]: value,
    });
  };

  const handleCandidateSelectionChange = (
    event: ChangeEvent<HTMLInputElement>
  ) => {
    setSelectedCandidate(event.target.value);
  };

  const handleTermsOfServiceOptionConfirmation = () => {
    setIsTermsCheckboxConfirmed(!isTermsCheckboxConfirmed);
  };

  return (
    <Container className="page-content">
      <VoterDetailsForm
        voterDetails={voterDetails}
        handleInputChange={handleVoterDetailChange}
      />
      <CandidatesSelection
        selectedCandidate={selectedCandidate}
        handleOptionChange={handleCandidateSelectionChange}
      />
      <SubmitVote
        isTermsCheckboxConfirmed={isTermsCheckboxConfirmed}
        handleChange={handleTermsOfServiceOptionConfirmation}
      />
    </Container>
  );
};

export default PageContent;
