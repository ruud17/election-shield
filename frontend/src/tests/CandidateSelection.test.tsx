import renderer from "react-test-renderer";
import CandidatesSelection from "../components/CandidatesSelection";
import { CandidatesSelectionProps } from "../components/CandidatesSelection";

const componentProps: CandidatesSelectionProps = {
  selectedCandidate: "Test Candidate",
  handleOptionChange: () => {},
};

test("renders correctly", () => {
  const tree = renderer
    .create(<CandidatesSelection {...componentProps} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
