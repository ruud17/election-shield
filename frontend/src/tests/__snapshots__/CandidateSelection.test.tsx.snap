// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="candidate row"
>
  <div
    className="fade alert alert-info show"
    role="alert"
  >
    Please select your preferred presidential candidate
  </div>
  <div
    className="col"
  >
    <div>
      <div
        className="bg-light text-black card"
      >
        <img
          className="card-img-top"
          src="Donald.jpeg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            <div
              className="form-check"
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                onChange={[Function]}
                type="radio"
                value="Donald Trump"
              />
              <label
                className="form-check-label"
                title=""
              >
                Donald Trump
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col"
  >
    <div>
      <div
        className="bg-light text-black card"
      >
        <img
          className="card-img-top"
          src="Joe.jpeg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            <div
              className="form-check"
            >
              <input
                checked={false}
                className="form-check-input"
                disabled={false}
                onChange={[Function]}
                type="radio"
                value="Joe Biden"
              />
              <label
                className="form-check-label"
                title=""
              >
                Joe Biden
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
